src/oauth2c.erl

```diff
@@ -137,7 +95,7 @@ retrieve_access_token(Type, Url, ID, Secret, Scope) ->
     ID      :: binary(),
     Secret  :: binary(),
     Scope   :: binary() | undefined,
-    Options :: list().
+    Options :: options().
```

Comment:
Actually this is wrong, options is not a proplist. check restc options. The type definition is the following:

```
-type options()      :: [option()].
-type option()       :: {atom(), term()} | atom().
```

So you can't assume it's a proplist. I suggest you use an atom instead, like `cache_tokens`.

Reply:
I understand the problem. Just to clarify; when you say use an atom instead, do you mean that I should simply pass the atom `cache_tokens` inside of a list? 

Also, isn't there a problem in `oauth2c:retrieve_access_token` as well since its Options argument is of type `proplist()` which is later passed to the Options argument of `oauth2c:request`?

Lastly, how come `options()` have type `proplist()` in oauth2c but a different type in restc? It appears to me that the Options variable (which has type `options()` in `oauth2c:request` and type `list()` in `oauth2c:retrieve_access_token`) is not used internally by oauth2c but simply forwarded to the restc module in the `restc:request` method. Does it not make more sense that the `options()` type would have the same type in both modules?

Reply:
Yes just the `cache_tokens` or any other name you think it's appropriate.

Yes. I think our `options` should have the same type as restc. I suspect the type definitions were wrong before.

Reply:
Actually, a proplist allows atoms and tuples. An atom `atom` is the shorthand for `{atom, true}`.
And to be even more pedantic, a proplist can contain any elements, but the `proplist` functions ignore anything that is not an atom or a two-tuple.

Reply:
uhhhhh that's quite cool. I didn't know that. Then it's correct, no changes needed.

Reply:
Thanks for the info Håkan! Is this issue resolved then @alexandre-kivra? Or should we also change the type of `options()` in oauth2c to the type that `options()` have in restc?